### ISSUE TITLE
Changing data onRest

### ISSUE THREAD

**Issue by jaredtibs** [nkbt/react-motion-loop#22]
Hi,

I'm having trouble clearly seeing how to insert logic at the end of a loop. For instance, changing the copy of a rotating header.

The loop breaks when I add an `onRest` prop, which you warn against in the README. How else does one accomplish this then? Or is that functionality not intended in this package?

Any help would be much appreciated. And apologies in advance if its right under my nose.

**Comment by xsteadybcgo** [comment]
I have  the same need that i want change the img tag's src attribute after every cycle loop，to make it looks like gif。

**Comment by xsteadybcgo** [comment]
> Hi,
> 
> I'm having trouble clearly seeing how to insert logic at the end of a loop. For instance, changing the copy of a rotating header.
> 
> The loop breaks when I add an `onRest` prop, which you warn against in the README. How else does one accomplish this then? Or is that functionality not intended in this package?
> 
> Any help would be much appreciated. And apologies in advance if its right under my nose.

I suggest to use react-motion instead of the react-motion-loop.
Blow is my app code, the important tip is that defaultState should be diffrent from the updateState
`import React, { Component, Fragment } from 'react';
import { Motion, spring } from 'react-motion'
import {
  LImg,
  RImg,
} from '../style';

class AlternatingImg extends Component {
  constructor(props) {
    super(props)
    this.state = {
      style: { left: 0, right: 0, opacity: 1 },
      onRest: undefined,
      aImg: props.aImg_1,
      bImg: props.bImg_1
    }
  }
  componentDidMount() {
    let count = 0;
    const { aImg_1, aImg_2, bImg_1, bImg_2 } = this.props
    this.timer = setInterval(() => {
      count++;
      this.setState({
        style: { left: spring(-20), right: spring(-20), opacity: spring(0) },
        onRest: () => {
          requestAnimationFrame(() => {
            this.setState({
              style: { left: spring(0), right: spring(0), opacity: spring(1) },
              aImg: count % 2 === 0 ? aImg_1 : aImg_2,
              bImg: count % 2 === 0 ? bImg_1 : bImg_2
            })
          })
        }
      })
    }, 4000)
  }
  componentWillUnmount() {
    clearInterval(this.timer)
  }
  render() {
    const {
      style,
      onRest,
      aImg,
      bImg
    } = this.state
    return (
      <Motion
        style={style}
        onRest={onRest}
      >
        {({ left, right, opacity }) => {
          return < Fragment >
            <LImg
              src={aImg}
              style={{ left, opacity }}
              alt="xxx" />
            <RImg
              src={bImg}
              style={{ right, opacity }}
              alt="xxx" />
          </Fragment>
        }}
      </Motion>
    );
  }
}

export default AlternatingImg;`

**Comment by nkbt** [comment]
Nah, not really, it is a simple library that uses `onRest` by itself. If you want to do something with `onRest`, do use `ReactMotion` directly instead.
Look at the source code, copy it and update logic to what you want.
